src/util/psbt/error.rs

```diff
@@ -18,6 +18,16 @@ use std::fmt;
 use blockdata::transaction::Transaction;
 use util::psbt::raw;
 
+use hashes::{self, sha256, hash160, sha256d, ripemd160};
+
+/// Support hash-preimages in psbt
+#[derive(Debug)]
+pub enum PsbtHash{
```

Comment:
nit: According to rust API guidelines , we should not use something like `psbt::PsbtHash`, so I propose to name this enum `HashKeys`

Reply:
Tbh, I find the current name way more intuitive.

Reply:
I think we should keep the current name, and I think that the rust API guidelines should make an exception for the word `Hash` which is already overloaded a zillion times :P

Reply:
So, I had to remove this enum completely in order to support a generic function because then I would need to somehow construct a concrete error type from a generic parameter. 

So, now the question is having the macro worth having if get slightly detailed error types? Or should I use go String return type instead.(as implemented in the latest revision).